src/screens/Resources/HomeScreen.js

```diff
@@ -0,0 +1,167 @@
+import React, {
+  useState, useEffect, useCallback,
+} from 'react';
+import {
+  ScrollView, Image, View, StyleSheet,
+} from 'react-native';
+import auth from '@react-native-firebase/auth';
+import firestore from '@react-native-firebase/firestore';
+import PropTypes from 'prop-types';
+import {
+  Text, Button, ActivityIndicator,
+} from 'react-native-paper';
+import { NavigationEvents } from 'react-navigation';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { theme } from '../../style';
+import { collections, nav } from './variables';
+
+export default function HomeScreen(props) {
+  const { navigation } = props;
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [snapshot, setSnapshot] = useState(null);
+  const [initialLoad, setInitialLoad] = useState(true);
+
+  const loadScreen = useCallback((id, title) => {
+    setLoading(true);
+    firestore()
+      .collection(collections.subcategories)
+      .where('parent', '==', id)
+      .get()
+      .then((subcategorySnapshot) => {
+        if (!subcategorySnapshot.empty) {
+          navigation.push(nav.subcategories, { snapshot: subcategorySnapshot, header: title });
+        } else {
+          firestore()
+            .collection(collections.items)
+            .where('parent', '==', id)
+            .get()
+            .then((itemSnapshot) => {
+              navigation.push(nav.itemList, { snapshot: itemSnapshot, header: title });
+            })
+            .catch((error) => {
+              setErrorMessage(error.message);
+              setLoading(false);
+            });
+        }
```

Comment:
So this makes sense to me because we discussed how to design this feature, but someone new to this code might find it hard to parse these lines. Can we refactor? In fact `async` and `await` might help to make it easier to parse.